### ISSUE TITLE
Mozilla addon validation issues

### ISSUE THREAD

**Issue by astral-sa** [MasterOdin/salr-redux#70]
>Hello,
>
>
>Your add-on Something Awful Last Read Redux has been reviewed and one or more of of its versions have been disabled due to critical issues discovered.
>
>Details:
>This version did not pass the review because of the following issues:
>
>1) JQuery 1.* is no longer accepted. Please use JQuery 3.*
>We accept JQuery/JQuery-UI libraries downloaded from &#39;ajax.googleapis.com&#39;, &#39;jquery.com&#39; or  &#39;jqueryui.com&#39;; and used without any modification (downloaded and not copy pasted). Minified versions are better. (file-name change does not matter)
>I&#39;m sorry, but we cannot accept modified, re-configured or customized libraries.
>Please use non-custom versions of jquery-ui, as well as the stock version of jquery and others, that match the original checksums on the maintainer&#39;s website.
>e.g.:
>js/jquery/jquery.js
>

Fair enough, easy enough. We're moving away from jQuery in general so it's not a huge roadblock.

If we do want to upgrade to jQuery 3.0, we can use jQuery migrate in two steps:
- [x] 1. Migrate older jQuery code to jQuery 1.12.x, optionally with jquery-migrate v1.4.1. We're already at 1.10.1 so that shouldn't be a big deal at all.
- [x] 1.5. Do we need to update jQuery-UI too? We have v1.10.3, which is on the allowed list. Do we need to check into cross-compatibilities between jQuery and jQuery UI? Updated it to 1.11.4 locally; when I tried 1.12 it made our edit note dialog look funny so I settled on 1.11.4 for now in the interests of getting stuff out fast.
- [x] 2. Once that's stable, we can use jquery-migrate v3.0.1 to migrate to jQuery v3.

ref: https://github.com/jquery/jquery-migrate/

>2) Add-on contains duplicate/hidden/unused/unnecessary files or folders. These complicate the review process and may contain sensitive information about the system you created the XPI on. Such files may also create cause for rejection. Please correct your packaging process so that these files aren&#39;t included.
>e.g.:
>images/emoticons/createJsonList.py
>js/jquery/colorpicker.old.js

Easy enough. Probably not worth disabling the extension over, though?
- [x] Probably just move createJsonList out of extension dir if it's build-related
- [x] colorpicker.old.js isn't referenced by anything and is just a duplicate of colorpicker.js, should be safe to just remove?

>3) Remote resources (eg images, fonts, CSS, etc) can be used for tracking (it will be allowed if the resource is from the same domain as the page the user is on). Please include them in the addon. e.g.:
>settings.html#L5-7, 71, 81, 101, 332, 379
>
>Please fix them and submit again. Thank you.

The settings page uses the site's own CSS to style some of the highlighting preview options. How dare SA be able to "track" SA members who are changing their settings.

Easy enough to move a few of the images used in the preview into the extension, but jeez. Again, this probably wasn't worth disabling the addon over.

- [X] Stylesheets: already fixed in 5a504c6
- [x] Images

>Please note the following for the next update:
>1) Unsafe assignment to innerHTML
>Due to both security and performance concerns, this may not be set using dynamic values which have not been adequately sanitized. This can lead to security issues or fairly serious performance degradation. e.g:
>js/salr.js - line: 1352, 1362, 1907

- [X] Fixed in 1a0b962

>2) This version contains obfuscated, minified, concatenated or otherwise machine-generated code. Please provide the original sources, together with instructions on how to generate the final XPI. Source code must be provided as an archive and uploaded using the source code upload field, which can be done during submission or on the version page in the developer hub.
>
>Please read through the instructions at https://developer.mozilla.org/en-US/Add-ons/Source_Code_Submission.
>More Information: https://blog.mozilla.org/addons/2016/04/05/improved-review-time-with-links-to-sources/
>
>Established libraries (eg: JQuery, bootstrap) do not require the inclusion of their sources.
>Please use non-custom versions of jquery-ui, as well as the stock version of jquery and others, that match the original checksums on the maintainer&#39;s website.

Yeah, I knew about this one and it was on the to-do list. Quick Reply seems to have some minified code from the forums JS - I have the original forums JS in my repo, so I can work on replacing it with the original JS + check what modifications have been made to it in SALR Redux via git blame or so. This is probably the one legitimate complaint because minified code could be doing anything and it's not fair to ask reviewers to try to understand it.

- [x] Replace modified minified paste handling code with modified original forums JS - fixed in  b105562

>Version(s) affected and disabled:
>2.1.8.12, 2.1.8.13a1, 2.1.8.13a2, 2.1.8.13a3, 2.1.8.13a4, 2.1.8.13a5, 2.1.8.13a6, 2.1.8.13a7, 2.1.8.13a8
>
>Reviewer:
>erosman
>
>You can resolve this by addressing the issues raised in the reviewer's notes and uploading a new version at https://addons.mozilla.org/en-US/developers/addon/salr-redux/versions

So we shall.

**Comment by astral-sa** [comment]
@MasterOdin I've fixed the remaining stuff as well as I could locally and pushed it to https://github.com/astral-sa/salr-redux/commits/master - didn't want to push the jquery update to master here in case it caused problems. Things seem to work but of course it's going to need testing. I pushed a build to the Firefox users to get some data in; it was approved automatically, then after I reached out to him, the human reviewer who disabled the previous versions gave it a quick look-over and said it looked good.
